### DISCUSSION TITLE
Is private repo support possible?

### DISCUSSION THREAD

**Discussion by antulik** [giscus/giscus#291] [Q&A] [answered]
Hi, I'm curious what's stopping giscus supporting private repositories? Is that a technical limitation of Github API or something else?

**Comment by laymonage** [comment, marked as answer]
Actually I guess that it currently should work just fine for private repositories, but:

- Anyone can see the discussion if they're not signed in (because giscus authenticates as the app, and the app has access to the repo).
- If they sign in and they have access to the repository, it should work. Otherwise, an error will show up (or maybe it keeps retrying?), but I personally haven't tested it.

**Reply by antulik** [reply]
Thanks that answers my question. 

Currently it doesn't work correctly with private repos. Giscus can't load existing comments, but it can create a new discussion.

Added https://github.com/giscus/giscus/issues/292
